account.go

```diff
@@ -6,81 +6,174 @@ import (
+	PayoutIntervalMonthly PayoutInterval = "monthly"
+	PayoutIntervalWeekly  PayoutInterval = "weekly"
+)
 
-	// Day is a constant value representing a daily payout interval.
-	Day Interval = "daily"
+const (
+	AccountRejectReasonFraud          string = "fraud"
+	AccountRejectReasonOther          string = "other"
+	AccountRejectReasonTermsOfService string = "terms_of_service"
+)
 
-	// Individual is a constant value representing an individual legal entity
-	// type.
-	Individual LegalEntityType = "individual"
+// AccountParams are the parameters allowed during account creation/updates.
+type AccountParams struct {
+	Params                    `form:"*"`
+	BusinessName              *string                       `form:"business_name"`
+	BusinessPrimaryColor      *string                       `form:"business_primary_color"`
+	BusinessURL               *string                       `form:"business_url"`
+	Country                   *string                       `form:"country"`
+	DebitNegativeBalances     *bool                         `form:"debit_negative_balances"`
+	DefaultCurrency           *string                       `form:"default_currency"`
+	Email                     *string                       `form:"email"`
+	ExternalAccount           *AccountExternalAccountParams `form:"external_account"`
+	FromRecipient             *string                       `form:"from_recipient"`
+	LegalEntity               *LegalEntityParams            `form:"legal_entity"`
+	PayoutSchedule            *PayoutScheduleParams         `form:"payout_schedule"`
+	PayoutStatementDescriptor *string                       `form:"payout_statement_descriptor"`
+	ProductDescription        *string                       `form:"product_description"`
+	StatementDescriptor       *string                       `form:"statement_descriptor"`
+	SupportEmail              *string                       `form:"support_email"`
+	SupportPhone              *string                       `form:"support_phone"`
+	SupportURL                *string                       `form:"support_url"`
+	TOSAcceptance             *TOSAcceptanceParams          `form:"tos_acceptance"`
+	Type                      *string                       `form:"type"`
+}
 
-	// IdentityVerificationPending is a constant value indicating that identity
-	// verification status is pending.
-	IdentityVerificationPending IdentityVerificationStatus = "pending"
+// LegalEntityParams represents a legal_entity during account creation/updates.
+type LegalEntityParams struct {
+	AdditionalOwners []*AdditionalOwnerParams `form:"additional_owners,indexed"`
 
-	// IdentityVerificationUnverified is a constant value indicating that
-	// identity verification status is unverified.
-	IdentityVerificationUnverified IdentityVerificationStatus = "unverified"
+	// AdditionalOwnersEmpty can be set to clear a legal entity's additional
+	// owners.
+	AdditionalOwnersEmpty bool `form:"additional_owners,empty"`
```

Comment:
@brandur-stripe While updating the examples in the docs I remembered I forgot to ask you about this one! That is (I think) the only place left in the entire library where we have an `empty` logic because `AdditionalOwners` is a slice of pointers. Do you know of a good fix or do we leave this as an exception for now?

Reply:
One possibility is that we could have users explicitly set an empty slice:

``` go
AdditionalOwners: []*AdditionalOwnerParams{},
```

Then in either a custom `AppendTo` or even directly in `form` (given that emptying an array with an empty string is probably a reusable convention) we could detect the slice's zero length and encode it as an empty string. It's a little weird, but I think it might be slightly better than having the companion `AdditionalOwnersEmpty`. What do you think?

Reply:
Sorry missed that one. Both would work. Since this is going away in X weeks though it might be easier to just leave as is. What do you think is better?

Reply:
Yeah good point. Let's just leave it for now then.